app/app/src/main/res/layout-v21/activity_main.xml

```diff
@@ -0,0 +1,168 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/activity_main"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:layout_margin="@dimen/activity_layout_margin"
+    android:background="@color/main_background"
+    android:orientation="vertical">
+
+    <TextView
+        android:id="@+id/calculation_label"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:background="@color/label_background"
+        android:text="@string/_0"
+        android:textSize="@dimen/activity_calculation_label_text_size"
+        app:layout_constraintBaseline_creator="@+id/activity_main"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent" />
+
+    <Button
+        android:id="@+id/clear_button"
+        android:layout_width="@dimen/activity_button_width"
+        android:layout_height="wrap_content"
+        android:text="@string/clear"
+        android:backgroundTint="@color/action_button_color"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@id/calculation_label" />
+
+    <Button
+        android:id="@+id/number_one"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/_1"
+        android:backgroundTint="@color/button_color"
+        app:layout_constraintEnd_toStartOf="@id/number_two"
+        app:layout_constraintTop_toBottomOf="@id/calculation_label" />
+
+    <Button
+        android:id="@+id/number_two"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/_2"
+        android:backgroundTint="@color/button_color"
+        app:layout_constraintEnd_toStartOf="@id/number_three"
+        app:layout_constraintTop_toBottomOf="@id/calculation_label" />
+
+    <Button
+        android:id="@+id/number_three"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/_3"
+        android:backgroundTint="@color/button_color"
```

Comment:
Take a look, you use the same width, height and backgroundTint for all number buttons, maybe you can use a "style" to re-use this properties.